File: back_api/src/main/java/com/example/api/entity/Members.java
package com.example.api.entity;

import jakarta.persistence.*;
import lombok.*;

import java.util.HashSet;
import java.util.Set;

@Entity
@Getter
@Builder
@AllArgsConstructor
@NoArgsConstructor
@ToString
@Table(name = "members")
public class Members extends BasicEntity{
  @Id
  @GeneratedValue(strategy = GenerationType.IDENTITY)
  private Long mid;
  @Column(unique=true, nullable = false)
  private String email;
  private String password;
  private String nickname;
  private String name;
  private String mobile;
  private Boolean fromSocial;

  @ElementCollection(fetch = FetchType.LAZY)
  @Builder.Default
  private Set<MembersRole> roleSet = new HashSet<>();

  public void addMemberRole(MembersRole membersRole) {
    roleSet.add(membersRole);
  }
}

File: front-journal/src/types/members.ts
export interface MembersDTO {
  mid: number
  email: string
  name: string
  nickname: string
  mobile: string
  fromSocial: boolean
  regDate: string
  modDate: string
}

export interface MembersRegisterRequest {
  email: string
  password: string
  name: string
  nickname: string
  mobile: string
  fromSocial: boolean
}
